testing/test_assertrewrite.py

```diff
@@ -1444,3 +1441,53 @@ def test_simple():
         )
         result = testdir.runpytest()
         result.assert_outcomes(passed=1)
+
+
+@pytest.mark.parametrize(
+    ("src", "expected"),
+    (
+        # fmt: off
+        pytest.param(b"", {}, id="trivial"),
+        pytest.param(
+            b"def x(): assert 1\n",
+            {1: "1\n"},
+            id="assert statement not on own line",
+        ),
+        pytest.param(
+            b"def x():\n"
+            b"    assert 1\n"
+            b"    assert 1+2\n",
+            {2: "1\n", 3: "1+2\n"},
+            id="multiple assertions",
+        ),
+        pytest.param(
+            # changes in encoding cause the byte offsets to be different
+            "# -*- coding: latin1\n"
+            "def ÀÀÀÀÀ(): assert 1\n".encode("latin1"),
+            {2: "1\n"},
+            id="latin1 encoded on first line\n",
+        ),
+        pytest.param(
+            # using the default utf-8 encoding
+            "def ÀÀÀÀÀ(): assert 1\n".encode(),
+            {1: "1\n"},
+            id="utf-8 encoded on first line",
+        ),
+        pytest.param(
```

Comment:
Could we add another case where we break the expression itself in multiple lines?

```python
        pytest.param(
            b"def x():\n"
            b"    assert (\n"
            b"        1 + \\"
            b"        2  # comment\n"
            b"    )\n",
            ...
        )
```

Also I commonly do asserts like this:

```python
assert obtained == [
    ...
] 
```

So another one like this might be in order:


```python
        pytest.param(
            b"def x():\n"
            b"    assert values == [\n"
            b"        1,\n"
            b"        2,\n"
            b"    )\n",
            ...
        )
```

Another case that comes to mind is assertions with a message:

```python
        pytest.param(
            b"def x():\n"
            b"    assert 1 + 1, 'some message'\n"
            ...
        )
```

and

```python
        pytest.param(
            b"def x():\n"
            b"    assert 1 + 1, f()\n"
            ...
        )
```

Let me know if those make sense or not.

Reply:
ooh, I'm definitely not handling messages right now -- should be an easy fix

the multi-line expression ones aren't all that different than the current cases but I can add those as well :+1:

Reply:
added a few of these, plus message-based ones and some tests for escaped newlines